### PR TITLE
Added replace module, for multiple regexp match + replace against entire file's contents

### PR DESCRIPTION
**Pull request by EvanK**
This implements a multiline regexp replacement against the _entire_ contents of a file. Unlike the `lineinfile` module, this would allow 1) matching more than one line of said  file and 2) replacing multiple matches. Currently, someone would have to implement this with a `shell` task using sed or perl.

The specific use case I built this for was to replace the entire `mpm_prefork` section from ubuntu's default apache config, while leaving the rest of the file in tact; this removes the section in question and replaces it with an include (to a file that I then created from a template):

``` yml
- replace: >
  dest=/etc/apache2/apache2.conf
  regexp='^(?:#[^\n]+\n)*<IfModule mpm_prefork_module>[^<]+?<\/IfModule>'
  replace='Include prefork.conf'
  backup=yes
  validate='/usr/sbin/apache2ctl -f %s -t'
```

I was also considering adding an `all` boolean or `count` numeric parameter to control how many replacements to perform against the file -- it's currently just replacing _all_ matches.

### CODE REVIEW

File: library/files/replace [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+    required: true
+    aliases: [ name, destfile ]
+    description:
+      - The file to modify.
+  regexp:
+    required: true
+    description:
+      - The regular expression to look for in the contents of the file.
+        Uses Python regular expressions; see
+        U(http://docs.python.org/2/library/re.html).
```
Comment (jarv) [inline library/files/replace:47]: might want to mention here that it defaults to using a multiline search

Comment (EvanK) [inline library/files/replace:47]: Done!

**Comment by mpdehaan** [comment]
Can you please resubmit this as a single commit so it is easier to review?

Thanks!

**Comment by EvanK** [comment]
Okay, squashed down to e88ab43

**Comment by jirutka** [comment]
+1

**Comment by gregbuehler** [comment]
+1

**Comment by jimi-c** [comment]
Merged, thanks!

**Pull request merged**